client/payments/data/types.ts

```diff
@@ -0,0 +1,32 @@
+export type PaymentGateway = {
+	id: string;
+	title: string;
+	description: string;
+	order: number | string;
+	enabled: boolean;
+	method_title: string;
+	method_description: string;
+	method_supports: Array< string >;
+	settings: Record< string, Record< string, string > >;
+};
+
+export type PluginsState = {
+	paymentGateways: Array< PaymentGateway >;
+	requesting: Record< string, boolean >;
+	errors: Record< string, RestApiError >;
+};
+
+interface RestApiErrorData {
+	status: number;
+}
+
+export type RestApiError = {
+	code: string;
+	data: Partial< RestApiErrorData >;
```

Comment:
I wonder if a partial is needed here. You can just make the RestApiErrorData type's properties optional instead.

e.g

```
interface RestApiErrorData {
  status?: number;
 }
 ```